lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1249,27 +1249,10 @@ def determine_group_level_monitored_resource_and_labels(tag,
       # Examples:
       # "container.<container_id>" // Docker container.
       # "k8s_pod.<namespace_name>.<pod_name>" // GKE pod.
-      if @enable_metadata_agent && local_resource_id
-        @log.debug 'Calling metadata agent with local_resource_id: ' \
-                  "#{local_resource_id}."
-        retrieved_resource = query_metadata_agent_for_monitored_resource(
+      if local_resource_id
+        converted_resource = convert_local_resource_id_to_monitored_resource(
           local_resource_id)
-        @log.debug 'Retrieved monitored resource from metadata agent: ' \
-                  "#{retrieved_resource.inspect}."
-        if retrieved_resource
-          resource = retrieved_resource
-          # TODO(qingling128): Fix this temporary renaming from 'gke_container'
-          # to 'container'.
-          resource.type = 'container' if resource.type == 'gke_container'
-        else
-          # TODO(qingling128): This entire else clause is temporary before we
-          # implement buffering and caching.
-          @log.warn('Failed to retrieve monitored resource from Metadata' \
-                     " Agent with local_resource_id #{local_resource_id}.")
-          constructed_k8s_resource = construct_k8s_resource_locally(
-            local_resource_id)
-          resource = constructed_k8s_resource if constructed_k8s_resource
-        end
+        resource = converted_resource if converted_resource
```

Comment:
[Optional] Since `resource` should be non-`nil` here, `resource &&= convert_local_resource_id_to_monitored_resource(local_resource_id)` would have worked as well, but your way is probably clearer.

Reply:
Yeah, i'd prefer the clearer way for readability.